elki-outlier/src/main/java/elki/outlier/density/HySortOD.java

```diff
@@ -0,0 +1,743 @@
+/*
+ * This file is part of ELKI:
+ * Environment for Developing KDD-Applications Supported by Index-Structures
+ * 
+ * Copyright (C) 2021
+ * ELKI Development Team
+ * 
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ * 
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package elki.outlier.density;
+
+import java.util.*;
+
+import elki.data.NumberVector;
+import elki.data.type.TypeInformation;
+import elki.data.type.TypeUtil;
+import elki.database.Database;
+import elki.database.datastore.DataStoreFactory;
+import elki.database.datastore.DataStoreUtil;
+import elki.database.datastore.WritableDoubleDataStore;
+import elki.database.ids.DBIDIter;
+import elki.database.relation.DoubleRelation;
+import elki.database.relation.MaterializedDoubleRelation;
+import elki.database.relation.Relation;
+import elki.logging.Logging;
+import elki.logging.progress.FiniteProgress;
+import elki.math.DoubleMinMax;
+import elki.outlier.OutlierAlgorithm;
+import elki.result.outlier.BasicOutlierScoreMeta;
+import elki.result.outlier.OutlierResult;
+import elki.result.outlier.OutlierScoreMeta;
+import elki.utilities.Alias;
+import elki.utilities.documentation.Description;
+import elki.utilities.documentation.Reference;
+import elki.utilities.documentation.Title;
+import elki.utilities.optionhandling.OptionID;
+import elki.utilities.optionhandling.Parameterizer;
+import elki.utilities.optionhandling.parameterization.Parameterization;
+import elki.utilities.optionhandling.parameters.IntParameter;
+
+/**
+ * Hypercube-Based Outlier Detection.
+ * <p>
+ * Algorithm that uses an efficient hypercube-ordering-and-searching strategy
+ * for fast outlier detection.
+ * Its main focus is the analysis of data with many instances and a
+ * low-to-moderate number of dimensions.
+ * <p>
+ * Reference:
+ * <p>
+ * Cabral, Eugênio F., Robson L.F. Cordeiro<br>
+ * Fast and Scalable Outlier Detection with Sorted Hypercubes<br>
+ * Proc. 29th ACM Int. Conf. on Information & Knowledge Management (CIKM'20)
+ * 
+ * @author Cabral, Eugênio F. (Original Code)
+ * @author Braulio V.S. Vinces (ELKIfication)
+ * @since 0.7.5
```

Comment:
The `@since` annotations at which release(!) a class was first available. We insert them automatically at a release. Hence on new contributions, this annotation should not be present.